Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -16,19 +16,41 @@ public class StarknetAccount: StarknetAccountProtocol {
         return StarknetSequencerInvokeTransaction(senderAddress: address, calldata: calldata, signature: signature, maxFee: params.maxFee, nonce: params.nonce)
     }
     
+    private func makeSequencerDeployAccountTransaction(classHash: Felt, salt: Felt, calldata: StarknetCalldata, maxFee: Felt, signature: [Felt]) -> StarknetSequencerDeployAccountTransaction {
+        return StarknetSequencerDeployAccountTransaction(
+            signature: signature,
+            maxFee: maxFee,
+            nonce: .zero,
+            contractAddressSalt: salt,
+            constructorCalldata: calldata,
+            classHash: classHash
+        )
+    }
+    
     public func sign(calls: [StarknetCall], params: StarknetExecutionParams) throws -> StarknetSequencerInvokeTransaction {
         let calldata = callsToExecuteCalldata(calls: calls)
         
         let sequencerTransaction = makeSequencerInvokeTransaction(calldata: calldata, signature: [], params: params)
         
-        let hash = TransactionHashCalculator.computeHash(of: sequencerTransaction, chainId: provider.starknetChainId)
+        let hash = StarknetTransactionHashCalculator.computeHash(of: sequencerTransaction, chainId: provider.starknetChainId)
         
         let transaction = StarknetInvokeTransaction(sequencerTransaction: sequencerTransaction, hash: hash)
         let signature = try signer.sign(transaction: transaction)
         
         return makeSequencerInvokeTransaction(calldata: calldata, signature: signature, params: params)
     }
     
+    public func signDeployAccount(classHash: Felt, calldata: StarknetCalldata, salt: Felt, maxFee: Felt) throws -> StarknetSequencerDeployAccountTransaction {
+        let sequencerTransaction = makeSequencerDeployAccountTransaction(classHash: classHash, salt: salt, calldata: calldata, maxFee: maxFee, signature: [])
+        
+        let hash = StarknetTransactionHashCalculator.computeHash(of: sequencerTransaction, chainId: provider.starknetChainId)
+        let transaction = StarknetDeployAccountTransaction(sequencerTransaction: sequencerTransaction, hash: hash)
+        
+        let signature = try signer.sign(transaction: transaction)
+        
+        return makeSequencerDeployAccountTransaction(classHash: classHash, salt: salt, calldata: calldata, maxFee: maxFee, signature: signature)
+    }
```

Comment:
I wonder if this could be simplified/refactored somehow. Correct me if I'm wrong, but it seems to me that the only difference between `StarknetSequencerDeployAccountTransaction` and `StarknetDeployAccountTransaction` is the `hash` field? Would it make sense to either just straigh away use `StarknetDeployAccountTransaction` and use it to sign, or maybe accepting sequencer transaction for signing?

If you think this is something that would take more time I have no problem postponing this change.